server/src/test/scala/io/delta/sharing/server/DeltaSharingServiceSuite.scala

```diff
@@ -106,13 +106,35 @@ class DeltaSharingServiceSuite extends FunSuite with BeforeAndAfterAll {
     readHttpContent(url, None, None, expectedTableVersion, "application/json; charset=utf-8")
   }
 
-  def readNDJson(url: String, method: Option[String] = None, data: Option[String] = None, expectedTableVersion: Option[Long] = None): String = {
-    readHttpContent(url, method, data, expectedTableVersion, "application/x-ndjson; charset=utf-8")
+  def readNDJson(
+    url: String,
+    method: Option[String] = None,
+    data: Option[String] = None,
+    expectedTableVersion: Option[Long] = None,
+    isStreamingQuery: Boolean = false): String = {
+    readHttpContent(
+      url,
+      method,
+      data,
+      expectedTableVersion,
+      "application/x-ndjson; charset=utf-8",
+      isStreamingQuery
+    )
   }
 
-  def readHttpContent(url: String, method: Option[String], data: Option[String] = None, expectedTableVersion: Option[Long] = None, expectedContentType: String): String = {
+
+  def readHttpContent(
+    url: String,
+    method: Option[String],
+    data: Option[String] = None,
+    expectedTableVersion: Option[Long] = None,
+    expectedContentType: String,
+    isStreamingQuery: Boolean = false): String = {
     val connection = new URL(url).openConnection().asInstanceOf[HttpsURLConnection]
     connection.setRequestProperty("Authorization", s"Bearer ${TestResource.testAuthorizationToken}")
+    if (isStreamingQuery) {
+      connection.setRequestProperty("User-Agent", "SparkStructuredStreaming")
```

Comment:
For the server to have different behavior based on the client(i.e., return metadata for queryTableChanges), in order to not break old client.

Reply:
@chakankardb Had a discussion with Ryan, it's not a good idea to leverage user-agent for different server behavior on RPC request, I'll add a parameter such as 'returnMetadata' for queryTableChanges, and update both Databricks and OSS server/client.

user-agent can still be used for usage tracking, and gate DFF (as it will be removed eventually).

I'll update this PR and send it later.

Reply:
@zsxwing Can you provide the reasoning here (for documentation)?

Reply:
`User-Agent` is usually informational and it's weird to use it as a parameter to request different results. In addition, `SparkStructuredStreaming` is a special system, but our network protocol should be designed for all systems.

Reply:
Another point Ryan mentioned is that for column mapping, we may also need to pass metadata to client to render dataframe correctly, which is not a streaming only use case.